media-gfx/pngnq/files/pngnq-1.1-gcc14-build-fix.patch

```diff
@@ -0,0 +1,33 @@
+Bug: https://bugs.gentoo.org/919210
+--- a/src/pngcomp.c
++++ b/src/pngcomp.c
+@@ -503,7 +503,7 @@ void printstats(struct statistics* stats, struct blockstats* bstats){
+   printf("Mean pixel color error: %f \n",stats->mean_error);
+   printf("Maximum pixel color error: %f \n",stats->max_error);
+   printf("Standard Deviation of error: %f\n",stats->stddev_error);
+-  printf("Image Dimensions %ld x %ld \n",image1_info.width,image1_info.height);
++  printf("Image Dimensions %u x %u \n",image1_info.width,image1_info.height);
```

Comment:
The correct type for `uint32_t` is PRIu32 macro found in [inttypes.h](https://en.cppreference.com/w/c/types/integer#Format_constants_for_the_fprintf_family_of_functions) header file.

```c
printf("Image Dimensions %"PRIu32" x %"PRIu32" \n",image1_info.width,image1_info.height);
```

Although in this case `png_uint_32` seems to *not* be a typedef to the `stdint.h` types. Just casting it to a type that's known to be big enough should be a portable way to handle cases like this:

```c
printf("Image Dimensions %lld x %lld \n", (long long int)image1_info.width, (long long int)image1_info.height);
```